notmuch/notmuch.c

```diff
@@ -1607,6 +1607,19 @@ char *nm_url_from_query(struct Mailbox *m, char *buf, size_t buflen)
   return buf;
 }
 
+/**
+ *  nm_query_window_available - Are windowed queries enabled for use?
+ */
+bool nm_query_window_available(void)
+{
+  const short c_nm_query_window_duration =
+      cs_subset_number(NeoMutt->sub, "nm_query_window_duration");
+  const bool c_nm_query_window_enable =
+      cs_subset_bool(NeoMutt->sub, "nm_query_window_enable");
+
+  return c_nm_query_window_enable || (c_nm_query_window_duration > 0);
```

Comment:
This is the opposite logic of the code it is replacing, but looks like you are fixing the pre-existing bug I pointed out on the previous commit.  Maybe call out the bug fix in the commit message?

Reply:
> This is the opposite logic of the code it is replacing

This logic is negated as this function checks if query windows **are** available while the original conditional checks if query windows **are not** available. The call sites use `!nm_query_window_available` to preserve logical equivalence. It's very common to separate out negations into the language keywords instead of function names. Prevents code that clumsy looking like "if not not_available()"

It also makes the logic easier to follow, "if explicitly enabled then available otherwise check the legacy enable condition (duration > 0)"

> you are fixing the pre-existing bug

There is no bug. If a user explicitly enables query windows then we won't check the duration for enable/disable condition.